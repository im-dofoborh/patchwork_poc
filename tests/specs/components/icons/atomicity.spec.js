import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import AtomicityIcon from '~/components/icons/Atomicity';

describe('AtomicityIcon component', () => {
  let router = null;
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('atomicity-icon', AtomicityIcon);
    localVue.use(VueRouter);
    router = new VueRouter();
    wrapper = mount(AtomicityIcon, {
      localVue,
      router,
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('uses fallback colour for invalid atomicity', () => {
    wrapper.setProps({
      atomicity: 'foo',
    })
    expect(wrapper.html()).toMatchSnapshot();
  });
});
