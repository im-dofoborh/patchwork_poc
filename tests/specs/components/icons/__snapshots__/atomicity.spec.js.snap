// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtomicityIcon component renders default variant 1`] = `
<a href="#/components/list?search=todo" class="">
  <div title="todo" class="sg-letter-icon" style="width: 30px;">
    <div class="sg-letter-icon__inner" style="height: 30px; width: 30px; background-color: rgb(11, 37, 51); color: rgb(255, 255, 255); font-size: 16.666666666666668px;">
      T
    </div>
  </div>
</a>
`;

exports[`AtomicityIcon component uses fallback colour for invalid atomicity 1`] = `
<a href="#/components/list?search=foo" class="">
  <div title="foo" class="sg-letter-icon" style="width: 30px;">
    <div class="sg-letter-icon__inner" style="height: 30px; width: 30px; background-color: rgb(33, 189, 255); color: rgb(255, 255, 255); font-size: 16.666666666666668px;">
      F
    </div>
  </div>
</a>
`;
